### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by Abhishekjha3515**
Added MACL(Model Aware Contrastive learning )  Loss Function in Loss folder

### CODE REVIEW

File: lightly/loss/macl_loss.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+class MACLLoss(nn.Module):
+    def __init__(self, temperature=0.1, alpha=0.5, A_0=0):
+        super().__init__()
+        self.t_0 = temperature
+        self.alpha = alpha
+        self.A_0 = A_0
```
Comment (guarin) [inline lightly/loss/macl_loss.py:10]:
```suggestion
    def __init__(self, temperature: float = 0.1, alpha: float = 0.5, A_0: float = 0):
        super().__init__()
        self.temperature = temperature
        self.alpha = alpha
        self.A_0 = A_0
```

Could you also add a short docstring that describes what the loss does, what the parameters are, and add a reference to the paper that introduced the loss? You can follow the example here: https://github.com/lightly-ai/lightly/blob/356ae56fd8f6d6bbc548d4d189d9444f780a62ca/lightly/loss/ntx_ent_loss.py#L17

**Review by guarin — CHANGES_REQUESTED** [review]
Thanks for the contribution and sorry for the late review!

Could you add a minimal test in `tests/loss/test_macl_loss.py`? It is enough to pass some dummy values and check that the loss returns a valid value.

File: lightly/loss/macl_loss.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        # Concatenate embeddings
+        out = torch.cat([z0, z1], dim=0)
+        batch_size = z0.shape[0]
+        n_samples = len(out)
+        
+        # Compute similarity matrix
+        cov = out @ out.T
+        
+        # Get positive and negative pairs
+        mask = self.mask_correlated_samples(batch_size).to(out.device)
```
Comment (guarin) [inline lightly/loss/macl_loss.py:40]:
```suggestion
        mask = cov.new_ones(cov.shape, dtype=bool)
        mask.diagonal()[:] = False
        mask.diagonal(batch_size)[:] = False
        mask.diagonal(-batch_size)[:] = False
```
This is a bit more concise and guarantees that the mask is on the same device as the input tensors.

Comment (Abhishekjha3515) [reply lightly/loss/macl_loss.py]: thanks for the input , i was busy with my office work I will do all the changes u have  suggested by this weekend

File: lightly/loss/macl_loss.py (hunk 1 above)
Comment (Abhishekjha3515) [reply lightly/loss/macl_loss.py:10]: ok i will add the loss doc also
